tests/ord-service/tests/subscription_flow_test.go

```diff
@@ -645,6 +645,7 @@ func TestConsumerProviderFlow(stdT *testing.T) {
 		require.ElementsMatch(stdT, []string{destination.Name, destinationSecond.Name}, []string{destinationsFromResponse[0].Get("sensitiveData.destinationConfiguration.Name").String(), destinationsFromResponse[1].Get("sensitiveData.destinationConfiguration.Name").String()})
 		stdT.Log("Successfully fetched system with bundles and destinations")
 
+		unassignFromFormation(stdT, ctx, providerApp.ID, "APPLICATION", consumerFormationName, secondaryTenant)
```

Comment:
Why this unassign had to be added here?

Reply:
The application in the test is assigned to formation in secondary tenant so the scenario label that is created has the tenant field set to the secondaryTenants ID and is not visible to subscriptionConsumerSubaccount.
When the unsubscribe is performed in the context of subscriptionConsumerSubaccount we attempt to delete the application. Before the fix the subscriptionConsumerSubaccountID did not see that the application is in formation and deleted it. With the fix it sees that there is a formation and fails to delete it. The unassign is needed before the unsubscribe so that the applcation can be deleted